packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -564,8 +584,26 @@ abstract class ImageProvider<T extends Object> {
   ///
   ///  * [ResizeImage], for modifying the key to account for cache dimensions.
   @protected
+  @Deprecated(
+    'Implement loadBuffer for faster image loading. '
+    'This feature was deprecated after v2.13.0-1.0.pre.',
+  )
   ImageStreamCompleter load(T key, DecoderCallback decode);
 
+  /// Converts a key into an [ImageStreamCompleter], and begins fetching the
+  /// image.
+  ///
+  /// The [decode] callback provides the logic to obtain the codec for the
+  /// image.
+  ///
+  /// See also:
+  ///
+  ///  * [ResizeImage], for modifying the key to account for cache dimensions.
+  @protected
+  ImageStreamCompleter loadBuffer(T key, DecoderBufferCallback decode) {
+    return _noOpCompleter;
```

Comment:
why not
```suggestion
    return load(key, PaintingBinding.instance.instantiateImageCodec);
```

And then ddrop `_noOpCompleter`.

Reply:
Done